### PR TITLE
Log warning when calculating time taken to load keys

### PR DESCRIPTION
**Pull request by usmansaleem**
## PR Description
Log warning instead of runtime exception if difference of time taken is negative i.e. system clock returned time in past.
## Fixed Issue(s)

#934 
## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

## Testing

- [x] I thought about testing these changes in a realistic/non-local environment.

### CODE REVIEW

File: signing/src/main/java/tech/pegasys/web3signer/signing/config/SignerLoader.java [hunk 1]
```diff
@@ -120,7 +120,12 @@ private MappedResults<SigningMetadata> convertConfigFileContent(
   }
 
   private static String calculateTimeTaken(final Instant start) {
-    return DurationFormatUtils.formatDurationHMS(Duration.between(start, Instant.now()).toMillis());
+    final Instant now = Instant.now();
+    final long timeTaken = Duration.between(start, now).toMillis();
+    if (timeTaken < 0) {
+      LOG.warn("System Clock returned time in past. Start: {}, Now: {}.", start, now);
+    }
+    return DurationFormatUtils.formatDurationHMS(Math.abs(timeTaken));
```
Comment (siladu) [inline signing/src/main/java/tech/pegasys/web3signer/signing/config/SignerLoader.java:128]: Think this could still throw the error when `timeTaken < 0`?

**Review by siladu — CHANGES_REQUESTED** [review]
Would be good to expose current bug in a unit test as well

File: signing/src/main/java/tech/pegasys/web3signer/signing/config/SignerLoader.java (hunk 1 above)
Comment (siladu) [reply signing/src/main/java/tech/pegasys/web3signer/signing/config/SignerLoader.java:128]: I see the Math.abs will avoid the error, but it will also mean we report inaccurate time taken so better not to report anything in this case

Comment (jframe) [reply signing/src/main/java/tech/pegasys/web3signer/signing/config/SignerLoader.java:128]: Agree ABS doesn't seem like the right solution. We should just not return anything in that case. So maybe just the return type to be Optional<String> and return the empty in the invalid case.

Comment (usmansaleem) [reply signing/src/main/java/tech/pegasys/web3signer/signing/config/SignerLoader.java]: addressed.

File: signing/src/main/java/tech/pegasys/web3signer/signing/config/SignerLoader.java [hunk 2]
```diff
@@ -42,6 +43,7 @@
 import org.apache.commons.lang3.time.DurationFormatUtils;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
+import org.jetbrains.annotations.VisibleForTesting;
```
Comment (siladu) [inline signing/src/main/java/tech/pegasys/web3signer/signing/config/SignerLoader.java:46]: nit: other code uses `import com.google.common.annotations.VisibleForTesting`

File: signing/src/test/java/tech/pegasys/web3signer/signing/config/SignerLoaderTest.java [hunk 3]
```diff
@@ -293,6 +296,20 @@ void callingLoadTwiceOnlyLoadSignersFromModifiedConfigFiles() throws IOException
...
 
+  @Test
+  void calculateTimeTakenWithFutureTimeShouldReturnEmpty() {
+    final Optional<String> timeTaken =
+        SignerLoader.calculateTimeTaken(Instant.now().plus(5, ChronoUnit.MINUTES));
+    assertThat(timeTaken).isEmpty();
+  }
+
+  @Test
+  void calculateTimeTakenWithPastTimeShouldReturnValue() {
```
Comment (siladu) [inline signing/src/test/java/tech/pegasys/web3signer/signing/config/SignerLoaderTest.java:307]: nit: maybe boundary case of same time too?

**Review by siladu — APPROVED** [review]

File: signing/src/main/java/tech/pegasys/web3signer/signing/config/SignerLoader.java (hunk 2 above)
Comment (usmansaleem) [reply signing/src/main/java/tech/pegasys/web3signer/signing/config/SignerLoader.java]: good catch ... fixed.
